shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -431,6 +431,13 @@ public static Event readFrom(DataInput in, int length) throws IOException {
             in.readFully(msg);
             return new Event(wrappedBuffer(msg));            
         }
+        
+        public ByteBuf getAsByteBuf() {
+            ByteBuf header = Unpooled.buffer(TYPE_PLUS_LENGTH_SIZE, TYPE_PLUS_LENGTH_SIZE);
+            header.writeInt(type.getCode());
+            header.writeInt(data.readableBytes());
+            return Unpooled.wrappedBuffer(header, data);
```

Comment:
`Unpooled.wrappedBuffer(...)` expects the user of caller to invoke `ByteBuf.release()`
> The caller is responsible for releasing this buffer.

Reply:
It doesn't matter. We are using unpooled buffers so the re-counting is a no-op. Unpooled buffers get deallocated by GC just like normal byte arrays. 
However you are correct that this would cause a problem IF we were to switch to using pooled buffers. Fortunately the integration tests are already configured to run with the leak detector turned on, so if we were to have enabled pooling and did not catch this those tests would have failed. 
To further prevent this from ever being an issue, I have added the decrement count when the event gets acked, and turned on the leak checker on the tests for the EventStreamWriter and the SegmentOutputStreamImpl. So if we ever switch to using pooled buffers it SHOULD be fine, and it it's not the unit tests will fail right away.